### PR TITLE
feat(client): add the base of the game

### PR DESCRIPTION
**Pull request by josephinecr**
## Checklist

- [ ] Tests
  - [x] I have tested these changes.
  - [ ] I have added automated tests (if applicable).
  - [x] I have tested on Linux.
  - [ ] I have tested on Windows.
- [ ] Documentation
  - [x] I have documented my functions with Doxygen comments.
  - [ ] I have updated the Wiki (if needed).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -18,22 +18,24 @@ set(RTYPE_SRC
...
+        src/RType/Components/Window.hpp
 )
 
 set(RTYPE_HEADERS
         include/Structures.hpp
         include/ECS.hpp
 )
 
-link_libraries(inih::inireader spdlog::spdlog)
+link_libraries(inih::inireader spdlog::spdlog sfml-system sfml-window sfml-graphics sfml-network sfml-audio)
```
Comment (Z4RM) [inline CMakeLists.txt:38]: Could not `target_link_libraries` be used for SFML libraries? The server doesn't needs them.

File: include/Structures.hpp [hunk 2]
```diff
@@ -50,4 +50,9 @@ struct Animation
     int frameRate;
 };
 
+struct String
+{
+    std::string s;
+};
+
```
Comment (Z4RM) [inline include/Structures.hpp:57]: What is the purpose of this structure?

File: src/RType/Components/Window.hpp [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+#pragma once
+#include <SFML/Graphics.hpp>
+#include "ECS.hpp"
+#include "Structures.hpp"
+
+struct RenderWindow
+{
+    sf::RenderWindow window;
+};
```
Comment (Z4RM) [inline src/RType/Components/Window.hpp:16]: What is the purpose of this structure?

File: src/RType/Components/Window.hpp [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+struct Mode
+{
+    sf::VideoMode mode;
+    Style style;
+};
+
+struct FrameLimit
+{
+    unsigned int limit = 60;
+};
```
Comment (Z4RM) [inline src/RType/Components/Window.hpp:38]: What is the purpose of this structure?

File: include/Structures.hpp (hunk 2 above)
Comment (MathisZUCCHERO) [reply include/Structures.hpp:73]: Stock the component String in componentManager.

File: src/RType/Components/Window.hpp (hunk 4 above)
Comment (MathisZUCCHERO) [reply src/RType/Components/Window.hpp:38]: Stock the frame limit data.

File: src/RType/Components/Window.hpp
[the remark this answers is not in the document]
Comment (MathisZUCCHERO) [reply src/RType/Components/Window.hpp:1]: Now RWindow is also use in RenderWindow.cpp

File: include/Structures.hpp (hunk 2 above)
Comment (Z4RM) [reply include/Structures.hpp:73]: Okay but what the String component is used for?

File: src/RType/Components/Window.hpp (hunk 3 above)
Comment (Z4RM) [reply src/RType/Components/Window.hpp]: "As String"?

File: src/RType/Components/Window.hpp (hunk 4 above)
Comment (Z4RM) [reply src/RType/Components/Window.hpp:38]: Okay, but why storing it in a structure, and not simply using an `uint` (`unsigned int`)?

File: src/RType/Components/Window.hpp
[the remark this answers is not in the document]
Comment (Z4RM) [reply src/RType/Components/Window.hpp:1]: Okay, so may be at least put in in `rtype::components`?

File: CMakeLists.txt [hunk 5]
```diff
@@ -48,4 +52,6 @@ target_compile_definitions(r-type_server PRIVATE RTYPE_IS_SERVER)
 add_executable(r-type_client ${RTYPE_SRC} ${RTYPE_HEADERS})
 
 target_compile_definitions(r-type_client PRIVATE RTYPE_IS_CLIENT)
+
+target_link_libraries(r-type_client sfml-system sfml-window sfml-graphics sfml-network sfml-audio)
```
Comment (Z4RM) [inline CMakeLists.txt:56]: Is `sfml-network` useful? We'll use Asio for communication between the client and the server.

File: conanfile.txt [hunk 6]
```diff
@@ -1,4 +1,5 @@
 [requires]
+xorg/system
```
Comment (Z4RM) [inline conanfile.txt:2]: What is `xorg/system` used for?

File: src/RType/RType.cpp [hunk 7]
```diff
@@ -17,5 +17,38 @@ int rtype::RType::run() {
...
+        {0, 0, 0},
+        {0, 0, 0},
+        {64, 64},
+        sprite2,
+        {"", 0, 0}
+        );
+
+    while (1) {
+        systemManager.updateSystems();
+    }
```
Comment (Z4RM) [inline src/RType/RType.cpp:53]: Infinie loop is a pretty bad idea...

File: CMakeLists.txt [hunk 8]
```diff
@@ -18,16 +18,24 @@ set(RTYPE_SRC
...
         src/RType/Config/Config.hpp
         src/RType/ModeManager/ModeManager.cpp
         src/RType/ModeManager/ModeManager.hpp
+        src/RType/Components/Window.cpp
+        src/RType/Components/Window.hpp
         src/Systems/Movement.cpp
         src/Systems/Movement.hpp
+        src/Systems/RenderWindow.cpp
+        src/Systems/RenderWindow.hpp
+        src/RType/RType.cpp
```
Comment (Z4RM) [inline CMakeLists.txt:35]: Warning, you added twice the same source.

File: CMakeLists.txt (hunk 5 above)
Comment (MathisZUCCHERO) [reply CMakeLists.txt:60]: Yeah, I imported all sfml libraries from another project.

File: CMakeLists.txt (hunk 8 above)
Comment (MathisZUCCHERO) [reply CMakeLists.txt:35]: I don't understand which ?

Comment (MathisZUCCHERO) [reply CMakeLists.txt:35]: Okay I see which one.

Comment (Z4RM) [reply CMakeLists.txt:35]: `src/RType/RType.cpp`

File: CMakeLists.txt (hunk 5 above)
Comment (Z4RM) [reply CMakeLists.txt:60]: So is it useful or not?

**Comment by Z4RM** [comment]
Is the wiki needs to be updated for these changes?

**Comment by Z4RM** [comment]
> Is the wiki needs to be updated for these changes?

@MathisZUCCHERO I'm asking this because it's in the checklist (https://github.com/Z4RM/Epitech-Tek3-CPP-R-Type/pull/23#issue-2725260544).

**Review by Z4RM — APPROVED** [review]
LGTM :+1:

**Pull request merged**